File: rust/71-simplify-path.rs
impl Solution {
    pub fn simplify_path(path: String) -> String {
        let mut v: Vec<String> = Vec::from("/");
        let mut curr = String::new();

        for c in str.chars() {
            curr.push(c);
            if c == '/' {
                if curr == "../" {
                    v.pop();
                }

                if curr != "./" || curr != '/' {
                    v.push(curr);
                }

                curr.clear();
            }  
        }

        if last == '/' && word.len()>1 {
            word.pop();
        }

        return word;
    }
}

File: tao/python/71-simplify-path.py
class Solution:
    def simplifyPath(self, path: str) -> str:
        arr = list(path.split("/"))
        ans = []

        for ele in arr:
            if not ele or ele=="." or (ele == ".." and not ans):
                continue
            if ele == "..":
                ans.pop()
            else:
                ans.append(ele)

        return '/'+'/'.join(ans)